tests/unit/pluginhandler/test_pluginhandler.py

```diff
@@ -1807,6 +1807,25 @@ def test_clean_prime_state(self):
         self.assertThat(self.handler.next_step(), Equals(steps.PRIME))
         self.assertFalse(os.path.exists(bindir))
 
+    def test_clean_prime_state_inconsistent_files(self):
```

Comment:
This test doesn't cover the [directory case](https://github.com/snapcore/snapcraft/pull/2450/files#diff-660a35e421f136410b0d3e5168c7a62cR1150). Mind either adding a new test, or also covering that here?

Reply:
The idea was to test removal of non-existent directories by passing directory "foo" (in line 1820 below), Effectively, when the test runs on an unpatched pluginhandler, `os.listdir()` raises `FileNotFoundError`.  Should I refactor to cover more cases/scenarios?

Reply:
Ha! _Totally_ missed `foo`. Yeah, just add a comment and I'm happy.

Reply:
Yeah, that is the reason I asked for the comment, hard to spot... maybe changing the name would also be good `inexistent-file` and `inexistent-dir` (or use nonexistent, both are valid according to Merriam Webster at least)

Reply:
Refactored test to change order and add comment as per Segio's suggestion. If good to go and nothing more to add, I'll squash the commits for merge.